src/sage/libs/singular/singular.pxd

```diff
@@ -29,6 +29,7 @@ cdef object   si2sa_ZZmod(number *n, ring *_ring, object base) noexcept
 cdef object   si2sa_NF(number *n, ring *_ring, object base) noexcept
 
 cdef object si2sa_intvec(intvec *v) noexcept
+cdef object si2sa_bigintvec(bigintmat *v) noexcept
```

Comment:
```suggestion
cdef object si2sa_bigintvec(bigintmat *v)
```

Also, after #37667 you'll need to adjust the context by removing the `noexcept` in lines 29, 31, 35.

The rule is: functions returning python objects cannot be `noexcept`. Cython 3.0 ignores it and starting with 3.0.9 it emits a warning; it could be an error in cython 3.1.